akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowMonitorSpec.scala

```diff
@@ -0,0 +1,70 @@
+package akka.stream.scaladsl
+
+import akka.stream.testkit.scaladsl.{ TestSource, TestSink }
+import akka.stream.{ ActorMaterializer, ActorMaterializerSettings }
+import akka.stream.testkit.AkkaSpec
+import akka.stream.FlowMonitor._
+
+import scala.concurrent.duration._
+
+class FlowMonitorSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+
+  implicit val materializer = ActorMaterializer(settings)
+
+  "A FlowMonitor" must {
+    "return Finished when stream is completed" in {
+      val ((source, monitor), sink) =
+        TestSource.probe[Any].monitorMat()(Keep.both).toMat(TestSink.probe[Any])(Keep.both).run()
+      source.sendComplete()
+      awaitCond(monitor.state == Finished, 3.seconds)
+      sink.expectSubscriptionAndComplete()
+    }
+
+    "return Finished when stream is cancelled from downstream" in {
+      val ((source, monitor), sink) =
+        TestSource.probe[Any].monitorMat()(Keep.both).toMat(TestSink.probe[Any])(Keep.both).run()
+      sink.cancel()
+      awaitCond(monitor.state == Finished, 3.seconds)
+    }
+
+    "return Failed when stream fails, and propagate the error" in {
+      val ((source, monitor), sink) =
+        TestSource.probe[Any].monitorMat()(Keep.both).toMat(TestSink.probe[Any])(Keep.both).run()
+      val ex = new Exception("Source failed")
+      source.sendError(ex)
+      awaitCond(monitor.state == Failed(ex), 3.seconds)
+      sink.expectSubscriptionAndError(ex)
+    }
+
+    "return Initialized for an empty stream" in {
+      val ((source, monitor), sink) =
+        TestSource.probe[Any].monitorMat()(Keep.both).toMat(TestSink.probe[Any])(Keep.both).run()
```

Comment:
you can use Source.maybe and Sink.ignore as well. Doesn't really matter though, just mentioning.